src/Rebing/GraphQL/Support/SelectFields.php

```diff
@@ -37,8 +38,22 @@ public function __construct(ResolveInfo $info, $parentType, array $args)
         if( ! is_null($info->fieldNodes[0]->selectionSet))
         {
             self::$args = $args;
+            $requestedFields = $info->getFieldSelection(5);
+
+            // If use pagination,
+            // $parentType & $requestedFields should use QueryType
+            // instead of paginationType
+            $paginationType = config('graphql.pagination_type', PaginationType::class);
+            if($info->returnType instanceof $paginationType){
+                foreach ($parentType->config['fields'] as $key => $field) {
```

Comment:
I don't think that works correctly, if the pagination also has other fields that are lists. For multiple fields the last type will always be selected

Reply:
I was thinking the data with pagination will always be the last one, if there is other fields that are lists most probably these fields are customPaginator fields and selectFields dont really need to care about these fields, as long as the paginate data be the last, then we can get the correct Type

Reply:
I change to another way to achieve this, should be OK right now